### ISSUE TITLE
[Bug]: 

### ISSUE THREAD

**Issue by clhiker** [babel/babel#14347]
### 💻

- [X] Would you like to work on a fix?

### How are you using Babel?

@babel/cli

### Input code

```js
async function* fn() {
  class A {
    [yield 1]() {}
  }

  class B extends A {
    [await 1]() {}
  }
}
```
Here is [REPL](https://babeljs.io/repl#?browsers=&build=&builtIns=false&corejs=false&spec=false&loose=false&code_lz=IYZwngdgxgBAZgV2gFwJYHsICp4QBQCUMA3gFAwxQA2oIMAgieRTANpioCmVAJjAIwBdQiQC-zcc2q0YAIRicAHsk4QedRmRZtgAd2CpkA4UWLiK48UA&debug=false&forceAllTransforms=false&shippedProposals=false&circleciRepo=&evaluate=false&fileSize=false&timeTravel=false&sourceType=unambiguous&lineWrap=true&presets=env%2Cflow&prettier=false&targets=&version=7.16.3&externalPlugins=&assumptions=%7B%7D)

### Configuration file name

babel.config.json

### Configuration

```json
{
  "sourceType": "unambiguous",
  "presets": ["@babel/preset-env"],
  "plugins": [
    "@babel/plugin-transform-runtime"
  ]
}
```

### Current and expected behavior

After transformed by babel, there is a bug ran by node
```bash
SyntaxError: missing ) after argument list
```

### Environment

- babel 
- - ├── @babel/cli@7.17.0=6
- - ├── @babel/generator@7.17.3
- - ├── @babel/plugin-transform-runtime@7.17.0
- - ├── @babel/preset-env@7.16.11
- Node: [v17.3.0]
- npm version [8.5.2]
- os [ubuntu 20.04]

**Comment by JLHwung** [comment]
This is a bug of `@babel/plugin-proposal-async-generator-functions`: Babel fails to transform the await expression in computed keys because the visitor skips in class methods.

To address this issue, we can import the `environmentVisitor` from "@babel/helper-environment-visitor" and merge the `environmentVisitor` with these visitors:

https://github.com/babel/babel/blob/a7b1181fef65b69a7ae6cd5e59045d311470e70a/packages/babel-plugin-proposal-async-generator-functions/src/index.ts#L10

https://github.com/babel/babel/blob/a7b1181fef65b69a7ae6cd5e59045d311470e70a/packages/babel-plugin-proposal-async-generator-functions/src/index.ts#L25

https://github.com/babel/babel/blob/a7b1181fef65b69a7ae6cd5e59045d311470e70a/packages/babel-helper-remap-async-to-generator/src/index.ts#L14

Here is an example about how to merge `environmentVisitor` with another visitor:

https://github.com/babel/babel/blob/a7b1181fef65b69a7ae6cd5e59045d311470e70a/packages/babel-helper-create-class-features-plugin/src/misc.ts#L6-L16

You can add a test in `packages/babel-plugin-proposal-async-generator-functions/test/fixtures/async-generators`.

Not related to this issue: I think the `environmentVisitor` should cover decorator expressions, too, which inherits the production parameter outside of the method itself, like computed keys.

To eventually fix all such issues, the `environmentVisitor` should be merged with all visitors that are meant to skip traversing `FunctionParent`.

In retrospect, this is an AST design issue, if we adopt the estree variant
```js
interface ClassMethod {
  key: Expression;
  value: Function;
}
```
the scope will be then associated to the `value` of `ClassMethod`, the `key` and `decorators` are clearly outside of the method scope, so we don't have to deal with this issue at all.

**Comment by Yokubjon-J** [comment]
I want to be assigned to the bug. I am about to start working on it.

**Comment by JLHwung** [comment]
The OP has selected "Would you like to work on a fix?", let's give he/she some time working on a fix.

**Comment by Yokubjon-J** [comment]
oh, i didn't notice it! ok!

**Comment by JLHwung** [comment]
@Yokubjon-J Since OP does not respond after 7 days. Do you still want to work on it?

**Comment by Yokubjon-J** [comment]
Yes, I will be happy to be assigned it!

**Comment by Yokubjon-J** [comment]
I added a test in the path specified and it passed. To recreate the OP's error I think I should also write the test in `babel\packages\babel-cli\test\fixtures\babel\[some-folder]` or `babel-cli\...\fixtures\babel-external-helpers\[some-folder]`. Is it ok if I create a folder containing the test file in the former directory?

**Comment by nicolo-ribaudo** [comment]
The test should go in `babel/packages/babel-plugin-proposal-async-generator-functions/test/fixtures`, since it's about the plugin behavior and not just about the CLI.

**Comment by Yokubjon-J** [comment]
Yep, I created the test (with a config file) and it passed, but I didn't get the OP's error.

**Comment by nicolo-ribaudo** [comment]
Could you open a draft PR with the test? So we can see why it's not failing.

**Comment by Yokubjon-J** [comment]
I suppose inserting `"@babel/helper-environment-visitor":"workspace:^"` and `"@babel/core":"workspace:^"` into `dependencies` section is right, isn't it?

**Comment by JLHwung** [comment]
`@babel/core` is already a peer deps of `proposal-async-generator-functions`. Adding `"@babel/helper-environment-visitor":"workspace:^"` should be sufficient.

**Comment by Yokubjon-J** [comment]
Whenever I include like so:   `const forAwaitVisitor = _core.traverse.visitors.merge([{...}, _helperEnvironmentVisitor])`, after running `make build` it changes to `const forAwaitVisitor = _core.traverse.visitors.merge([{...}, _helperEnvironmentVisitor.**default**])`, is this behaviour normal?
